linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -208,7 +208,7 @@ impl<F: Float, L: Label> TreeNode<F, L> {
 
         for i in 0..x.nrows() {
             if mask.mask[i] {
-                if x[(i, best_feature_idx)] < best_split_value {
+                if x[(i, best_feature_idx)] <= best_split_value {
```

Comment:
Isn't this an arbitrary choice?

Reply:
the equality was introduced to satisfy a test with only synthetic values. But `best_split_value` should actually be some kind of middle point between previous and next optimal split decision

Reply:
Ah okay I see

Reply:
the best_split_value is now the midpoint